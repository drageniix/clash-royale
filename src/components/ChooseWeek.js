import React from 'react'
import { setNewWeek } from '../redux/actions'
import { connect } from 'react-redux'

export class ChooseWeek extends React.Component {
    setNewWeek = index => e => {
        this.props.setNewWeek(this.props.lastWeeks, index)
    }

    render(){
        return (
            <span className="weekSelection">
                <a>History</a>
                <div className="weekSelection__weeks">
                    {this.props.lastWeeks.map((item, index) => (
                        <li className={(this.props.current === index ? "selectedWeek" : "")} 
                            key={index} 
                            onClick={this.setNewWeek(index)}>
                            {item.display}</li>
                    ))}
                </div>
            </span>
        )
    }

} 

const mapStateToProps = state => ({
    current: state.current,
    lastWeeks: state.api.lastWeeks
})

const mapDispatchToProps = dispatch => ({
    setNewWeek: (lastWeeks, current) => dispatch(setNewWeek(lastWeeks, current))
})

export default connect(mapStateToProps, mapDispatchToProps)(ChooseWeek)