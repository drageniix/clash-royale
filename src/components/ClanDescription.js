import React from 'react'
import { connect } from 'react-redux'

const ClanDescription = ({admin}) => (
    <section className="description">
        <div className="description__reqs">
            <h2>Clan Requirements</h2>
            <p><strong>Member:</strong> Donate at least {admin.minDonations} cards weekly and participate in {admin.minWarBattles}/10 war battles, missing a maximum of 1 war day battle when you participated in collection day.</p>
            <p><strong>Elder:</strong> Have at least {admin.minTrophies} trophies and donate more than {admin.elderDonations} cards. Must participate in at least {admin.elderWarBattles}/10 wars. Kicking members and sending invitations to those below basic member requirements requires permission.</p>
        </div>
    </section>
)

export default connect(state => ({
    admin: state.admin
}))(ClanDescription)