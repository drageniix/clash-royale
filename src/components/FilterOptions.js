import React from 'react';
import { getSelectedFilterClass } from '../redux/selectors';
import { setFilter } from '../redux/actions';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export class FilterOptions extends React.Component {
    onFilter = filterType => () => {
        this.props.onFilter(filterType);
    };

    render() {
        return (
            <div className="filter">
                <div className="filter__vaporware" />
                <ul className="filter__list">
                    <li
                        className={this.props.getSelectedFilterClass('none')}
                        onClick={this.onFilter('byNone')}
                    >
                        All Members
                    </li>
                    <li
                        className={this.props.getSelectedFilterClass(
                            'promotion'
                        )}
                        onClick={this.onFilter('byPromotion')}
                    >
                        Promotion
                    </li>
                    <li
                        className={this.props.getSelectedFilterClass(
                            'probation'
                        )}
                        onClick={this.onFilter('byProbation')}
                    >
                        Probation
                    </li>
                    <li
                        className={this.props.getSelectedFilterClass(
                            'demotion'
                        )}
                        onClick={this.onFilter('byDemotion')}
                    >
                        Demotion
                    </li>
                </ul>
            </div>
        );
    }
}

const mapStateToProps = state => ({
    getSelectedFilterClass: filter =>
        getSelectedFilterClass(state.filters.filter, filter)
});

const mapDispatchToProps = dispatch => ({
    onFilter: filterType => dispatch(setFilter[filterType]())
});

FilterOptions.propTypes = {
    getSelectedFilterClass: PropTypes.func.isRequired,
    onFilter: PropTypes.func.isRequired
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(FilterOptions);
