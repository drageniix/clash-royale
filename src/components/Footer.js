import moment from 'moment'
import React from 'react'
import { connect } from 'react-redux'

export const Footer = ({ time, discord }) => (
    <footer className="footer">
        <div className="footer__links">
            <a href={discord}><div className="icon--discord"></div></a>
        </div>
        <p className="footer__updated">Clan data last updated {time && moment.unix(time).fromNow()}.</p>
        <p className="footer__credits">@meliaesc</p>
    </footer>
)

export default connect(state => ({
    time : state.time,
    discord: state.discord
}))(Footer)