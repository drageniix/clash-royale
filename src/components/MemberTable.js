import React from 'react'
import { capitalizeRole, checkDonations } from '../utilities/members'
import { getMembers } from '../redux/selectors'
import { setQuery } from '../redux/actions'
import { connect } from 'react-redux'

export class MemberTable extends React.Component {
    onClick = memberTag => e => {
        this.props.onClick(memberTag)
    }

    render() {
        return (
            <tbody>
                {this.props.members.map((member, index) => (
                    <tr key={index} className={"clanMembers__" + (member.eligibleForPromotion ? "promotion" : member.onProbation ? "probation" : member.dangerOfDemotion ? "demotion" : "row")}>
                        <td className="align-left"> #{member.clanRank} ({member.trophies})</td>
                        <td className="align-right">{capitalizeRole(member)}</td>
                        <td className="pointer align-left member-link" onClick={this.onClick(member.tag)}>
                            {member.name}
                        </td>
                        <td className="align-right">{checkDonations(member)}</td>
                        <td className="align-right">{member.warBattles}{member.missedWarBattles ? (" (" + member.missedWarBattles + "M)") : ""}</td>
                    </tr>
                ))}
            </tbody>
        )
    }
}

const mapStateToProps = state => ({
    members: getMembers(state.api.members, state.filters)
})

const mapDispatchToProps = dispatch => ({
    onClick: query => {
        window.scroll({ top: 150, left: 0, behavior: 'smooth' }) //can't use in tests
        dispatch(setQuery(query))
    }
})

export default connect(mapStateToProps, mapDispatchToProps)(MemberTable)