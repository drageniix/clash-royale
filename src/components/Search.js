import React from 'react'
import { setQuery } from '../redux/actions'
import { connect } from 'react-redux'

export class Search extends React.Component {
    setQuery = e => {
        const query = e.target.value
        this.props.setQuery(query)
    }

    render(){
        return (
            <div className="form">
                <form className="form__content" onSubmit={() => false}>
                    <input onChange={this.setQuery} value={this.props.query} type="text" placeholder="Member Search"
                        id="search" className="form__input" spellCheck={false} autoFocus />
                    <button className="form__button">
                        <svg className="form__icon" viewBox="0 0 20 20">
                            <title>Search Members</title>
                            <path d="M17.545 15.467l-3.779-3.779c0.57-0.935 0.898-2.035 0.898-3.21 0-3.417-2.961-6.377-6.378-6.377s-6.186 2.769-6.186 6.186c0 3.416 2.961 6.377 6.377 6.377 1.137 0 2.2-0.309 3.115-0.844l3.799 3.801c0.372 0.371 0.975 0.371 1.346 0l0.943-0.943c0.371-0.371 0.236-0.84-0.135-1.211zM4.004 8.287c0-2.366 1.917-4.283 4.282-4.283s4.474 2.107 4.474 4.474c0 2.365-1.918 4.283-4.283 4.283s-4.473-2.109-4.473-4.474z"></path>
                        </svg>
                    </button>
                </form>
            </div>
        )
    }
}

const mapStateToProps = state => ({
    query: state.query
})

const mapDispatchToProps = dispatch => ({
    setQuery: query => dispatch(setQuery(query))
})

export default connect(mapStateToProps, mapDispatchToProps)(Search)