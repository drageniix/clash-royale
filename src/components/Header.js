import React from 'react'
import { connect } from 'react-redux'
import ChooseWeek from './ChooseWeek';

export const Header = ({ name, discord }) => (
    <header className="header">
        <div className="header__content">
            <div className="header__brand">
                <img className="header__brand--logo" src="./favicon.png" />
                <h1 className="header__brand--title">{name}</h1>
            </div>
            <div className="header__links">
                <ChooseWeek/>
                <a href={discord}>Discord</a>
                <a href="https://royaleapi.com/clan/PGVRPVG">All Stats</a>
            </div>
        </div>
    </header>
)

export default connect(state => ({
    name: state.api.clan.name,
    discord: state.api.discord
}))(Header)
