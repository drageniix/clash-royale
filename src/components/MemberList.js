import React from 'react'
import FilterOptions from './FilterOptions';
import SortOptions from './SortOptions';
import { capitalizeFirstLetter, checkDonations } from '../utilities/basic'
import { getMembers } from '../redux/selectors'
import { setQuery } from '../redux/actions'
import { connect } from 'react-redux'

export class MemberList extends React.Component {
    onClick = memberTag => e => {
        window.scroll({ top: 200, left: 0, behavior: 'smooth' }) //can't use in tests
        this.props.onClick(memberTag)
    }

    render() {
        return (
            <section className="filter">
                <FilterOptions />
                <table className="clanMembers">
                    <SortOptions />
                    {this.props.members.map((member, index) => (
                        <tr key={index} className={"clanMembers__" + (member.eligibleForPromotion ? "promotion" : member.onProbation ? "probation" : member.dangerOfDemotion ? "demotion" : "row")}>
                            <td className="align-left"> #{member.clanRank} ({member.trophies})</td>
                            <td className="align-right">{capitalizeFirstLetter(member.role)}</td>
                            <td className="pointer align-left member-link" onClick={this.onClick(member.tag)}>
                                {member.name}
                            </td>
                            <td className="align-right">{checkDonations(member)}</td>
                            <td className="align-right">{member.warBattles}{member.missedWarBattles ? (" (" + member.missedWarBattles + "M)") : ""}</td>
                        </tr>
                    ))}
                </table>
            </section>
        )
    }
}

const mapStateToProps = state => ({
    members: getMembers(state.api.members, state.filters)
})

const mapDispatchToProps = dispatch => ({
    onClick: query => dispatch(setQuery(query))
})

export default connect(mapStateToProps, mapDispatchToProps)(MemberList)