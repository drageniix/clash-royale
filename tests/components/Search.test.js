import React from 'react'
import { Search } from '../../src/components/Search';

test('Search: set query', () => {
    const dispatchSpy = jest.fn()
    const wrapper = shallow(<Search
        query="potato"
        setQuery={dispatchSpy} />)

    const query="meliaesc"
    wrapper.find('input').simulate('change', {
        preventDefault: () => { },
        target: { value: query }
    })

    expect(dispatchSpy).toHaveBeenLastCalledWith(query)
    expect(wrapper).toMatchSnapshot()
})