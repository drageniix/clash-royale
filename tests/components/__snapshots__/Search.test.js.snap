// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search: set query 1`] = `
<div
  className="form"
>
  <form>
    <input
      autoFocus={true}
      className="form__input"
      id="search"
      onChange={[Function]}
      placeholder="Member Search"
      spellCheck={false}
      type="text"
      value="potato"
    />
    <label
      className="form__label"
      htmlFor="search"
    >
      Member Search
    </label>
  </form>
</div>
`;
